Danielhw trigger docs (#41)
slackapi/deno-slack-api: docs/triggers/trigger-basics.md

### Trigger Types
There are currently 4 supported Trigger types, [Shortcut](shortcut-trigger.md), [Webhook](webhook-trigger.md), 
[Event](event-trigger.md), and [Scheduled](scheduled-trigger.md). Each Trigger type has it's own configuration object, however all Triggers have parameters that are common to all Trigger types, these common parameters are as follows:

| Parameter name  | Required?     | Description                                                          |
| ----------------|:-------------:| ---------------------------------------------------------------------|
| `type`            | Yes           | The type of Trigger (one of `shortcut`, `event`, `scheduled`, or `webhook`)|
| `name`            | Yes           | The name of the Trigger                                              |
| `description`     | No            | A description of the purpose of the Trigger                          |

> | `workflow`        | Yes           | Which workflow the Trigger connects to                               |

Reviewer: ```suggestion
| `workflow`        | Yes           | Which Workflow the Trigger connects to                               |
```

> | `inputs`          | No            | What inputs (defined in the manifest) are passed to the Trigger      |

Reviewer: ```suggestion
| `inputs`          | No            | What inputs (defined in the manifest) are passed to the Workflow      |
```

[Event](event-trigger.md), and [Scheduled](scheduled-trigger.md). Creation of each trigger type has a specific creation
object type, however all triggers have parameters that are common to all trigger types, these common parameters are as follows:

| Parameter name  | Required?     | Description                                                          |
| ----------------|:-------------:| ---------------------------------------------------------------------|
| `type`            | Yes           | The type of trigger (one of `shortcut`, `event`, `scheduled`, or `webhook`)|
| `name`            | Yes           | The name of the trigger                                              |
| `description`     | No            | A description of the purpose of the trigger                          |
| `workflow`        | Yes           | Which workflow the trigger connects to                               |
| `inputs`          | No            | What inputs (defined in the manifest) are passed to the trigger      |

In addition to these common parameters, specific trigger types will have their own unique parameters which can be viewed in 
their respective documents.

### Trigger Context Data

> Each trigger type has access to a data context object which includes information related to the context of the trigger activation. This data object can be used to fill the input fields of the trigger being activated. Each of the four trigger types has access to separate information which is contextual based on the trigger type, which will be provided in the specific article for the trigger type.

Reviewer: ```suggestion
Each Trigger type has access to a `data` context object which includes information related to the context of the Trigger activation. This `data` object can be used to define values for `inputs` and `filters` of the Trigger being activated. Each of the four Trigger types has access to separate information which is contextual based on the Trigger type. These details will be provided in the specific article for that Trigger type.
```
